hal/sleep_api.h

```diff
@@ -27,6 +27,42 @@
 extern "C" {
 #endif
 
+/**
```

Comment:
What is that? That should be on the branch already?

Reply:
I added requirements section since it wasn't available on this branch. This has been created based on function descriptions.

Reply:
That explains it, thanks @mprse